sw/host/opentitanlib/src/image/manifest.rs

```diff
@@ -104,12 +113,20 @@ impl Default for SigverifySpxSignature {
 
 /// Extension header.
 #[repr(C)]
-#[derive(AsBytes, FromBytes, FromZeroes, Debug, Default)]
+#[derive(AsBytes, FromBytes, FromZeroes, Debug, Default, Serialize, Deserialize)]
 pub struct ManifestExtHeader {
     pub identifier: u32,
     pub name: u32,
 }
 
+impl ManifestExtHeader {
+    pub fn write(&self, dest: &mut impl Write) -> Result<()> {
+        dest.write_u32::<LittleEndian>(self.identifier)?;
+        dest.write_u32::<LittleEndian>(self.name)?;
+        Ok(())
+    }
+}
+
```

Comment:
Since the `ManifestExtHeader` implements `AsBytes`, this write function isn't really necessary (except that it provides a nice regular interface for use in `ManifestExtIsfb::write`.

I'd change this function to use the `AsBytes` implementation so as to not have two binary conversion functions:
```
pub fn write(&self, dest: &mut impl Write) -> Result<()> {
  dest.write(self.as_bytes())?;
  Ok(())
}
```

Reply:
Done. I ended up having to use `dest.write_all(self.as_bytes())?;`. 

Tested changes.